src/actions/cart.ts

```diff
@@ -18,4 +22,12 @@ export const removeFromCart = (index: number) => {
     type: CartTypes.RemoveFromCart,
     index,
   });
+};
+
+export const checkoutCart = (history: History, checkoutData: CheckoutData) => {
+  //TODO: include a 'purchase' service call
+  history.push('/thankyou');
```

Comment:
@timchin thoughts on passing `history` into the action like this?

Reply:
I don't think that this is very common? Most action creators tend to be written to only create the actions without any side effects (such as changing the url). Creating an action doesn't necessarily mean that the action has been dispatched either!

As this expands, this requirement would likely solve itself, since often the checkout action would be tied to a network request tied to a thunk, and that leads to two states: successful/unsuccessful checkout. 

```js
dispatch(checkoutCart(checkoutData)).then(() => {
  history.push('/thankyou')
}).catch(() => 
   // show error
})
```

Reply:
Hey, @timjchin - generally speaking, where should side-effects be handled? Should they exclusively be handled in thunks that are in turn dispatched from the action creator?

I have a couple of `TODOs` about side-effects in the code; would love to know your thoughts.

Reply:
Hey, @timjchin - generally speaking, where should side-effects be handled? Should they exclusively be handled in thunks that are in turn dispatched from the action creator?

I have a couple of `TODOs` about side-effects in the code; would love to know your thoughts.

Reply:
This is done!